### PR TITLE
Docs on Runtime Errors

### LINKED ISSUE #709: Documentation should explain why runtime should not panic
There is no clear documentation around the idea that Substrate runtimes should be non-throwing, let alone documentation about how to ensure that this is the case. This should be added to the Runtime section in the Knowledge Base.

### PR DESCRIPTION
**Pull request by danforbes**
Closes #709

### CODE REVIEW

File: docs/knowledgebase/runtime/errors.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+In order to emit custom errors from a pallet, the pallet's module must configure the `Error` type.
+
+```rust
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		// Errors must be initialized if they are used by the pallet.
+		type Error = Error<T>;
+
+		/* snip */
```
Comment (riusricardo) [inline docs/knowledgebase/runtime/errors.md:41]:
```suggestion
		/* --snip-- */
```

Comment (riusricardo) [reply docs/knowledgebase/runtime/errors.md:41]: We kinda used a few variants of `snip` in order to be consistent. We are using this options depending on the file type:
```
// --snip--
/* --snip-- */
# --snip--
```

File: docs/knowledgebase/runtime/errors.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+[`Result` types](https://substrate.dev/rustdocs/v2.0.0/frame_support/dispatch/result/enum.Result.html).
+The `Result` enum type possesses an `Err` variant that allows a function to indicate that it failed
+to execute successfully without needing to panic. Dispatchable calls in the FRAME system for runtime
+development _must_ return a
+[`DispatchResult` type](https://substrate.dev/rustdocs/v2.0.0/frame_support/dispatch/type.DispatchResult.html)
+that _should_ be a
+[`DispatchError`](https://substrate.dev/rustdocs/v2.0.0/frame_support/dispatch/enum.DispatchError.html)
+if the dispatchable function encountered an error.
+
+Each FRAME pallet may define custom `DispatchError`s by using
```
Comment (riusricardo) [inline docs/knowledgebase/runtime/errors.md:18]:
```suggestion
Each FRAME pallet may define custom **DispatchError**s by using
```

Comment (riusricardo) [reply docs/knowledgebase/runtime/errors.md:18]: Just suggesting this because it won't add the extra space after the word.

**Review by riusricardo — APPROVED** [review]

**Review by wheresaddie — APPROVED** [review]

**Pull request merged**